ddpui/ddpairbyte/airbyte_service.py

```diff
@@ -632,79 +632,65 @@ def create_connection(
 
 def update_connection(
     workspace_id: str,
-    connection_id: str,
     connection_info: schema.AirbyteConnectionUpdate,
+    current_connection
 ) -> dict:
     """Update a connection of an airbyte workspace"""
 
     if not isinstance(workspace_id, str):
         raise HttpError(400, "workspace_id must be a string")
-    if not isinstance(connection_id, str):
-        raise HttpError(400, "connection_id must be a string")
     if len(connection_info.streams) == 0:
         error_message = f"must specify at least one stream workspace_id={workspace_id}"
         logger.error(error_message)
         raise HttpError(400, error_message)
 
     sourceschemacatalog = get_source_schema_catalog(
-        workspace_id, connection_info.sourceId
+        workspace_id, current_connection["sourceId"]
     )
 
-    payload = {
-        "connectionId": connection_id,
-        "sourceId": connection_info.sourceId,
-        "destinationId": connection_info.destinationId,
-        "sourceCatalogId": sourceschemacatalog["catalogId"],
-        "syncCatalog": {
-            "streams": [
-                # <== we're going to put the stream configs in here in the next step below
-            ]
-        },
-        "prefix": "",
-        "namespaceDefinition": "destination",
-        "namespaceFormat": "${SOURCE_NAMESPACE}",
-        "nonBreakingChangesPreference": "ignore",
-        "scheduleType": "manual",
-        "geography": "auto",
-        "name": connection_info.name,
-        "operations": [
-            {
-                "name": "Normalization",
-                "workspaceId": workspace_id,
-                "operatorConfiguration": {
-                    "operatorType": "normalization",
-                    "normalization": {"option": "basic"},
-                },
-            }
-        ],
-    }
+    # update the name
+    if connection_info.name:
+        current_connection["name"] = connection_info.name
+
+    # update the destination schema
+    if connection_info.destinationSchema:
+        current_connection["namespaceDefinition"] = "customformat"
+        current_connection["namespaceFormat"] = connection_info.destinationSchema
+
```

Comment:
and if no `destinationSchema` in (the new) `connection_info`, should we not reset the `destinationSchema` which is potentially in `current_connection`?